packages/loopring_v3.1/contracts/impl/modules/OffchainWithdrawalModule.sol

```diff
@@ -0,0 +1,105 @@
+/*
+
+  Copyright 2017 Loopring Project Ltd (Loopring Foundation).
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+pragma solidity ^0.5.11;
+pragma experimental ABIEncoderV2;
+
+import "../../iface/modules/IOffchainWithdrawalModule.sol";
+import "./helpers/WithdrawalModule.sol";
+
+
+/// @title  OffchainWithdrawalModule
+/// @author Brecht Devos - <brecht@loopring.org>
+contract OffchainWithdrawalModule is WithdrawalModule, IOffchainWithdrawalModule
+{
+    constructor(address exchangeAddress, address vkProviderAddress)
+        WithdrawalModule(exchangeAddress, vkProviderAddress, 0, 0)
+        public
+    {
+        // Nothing to do
+    }
+
+    function onRevert(
+        uint blockIdx
+        )
+        external
+        onlyExchange
+    {
+        // Nothing to do
+    }
+
+    function onRemove()
+        external
+        onlyExchange
+        returns (bool)
+    {
+        // This module can NEVER be removed (otherwise this module isn't authorized
+        // for withdrawing from the exchange anymore which users may still need to do at any time).
+        // Use disable() to stop this module from being used for future work.
+        return false;
```

Comment:
If this is true, we should really restrict that each factory can only register one module (instance). Otherwise if you register another instance of this module, you can not remove it.

Reply:
After a second thought, this comment may not be that important. As you said, maybe we allow multiple instances of the same module, although I don't have a use case in mind.